### ISSUE TITLE
Data views: The featured image should provide information about the featured image

### ISSUE THREAD

**Issue by afercia** [WordPress/gutenberg#59830]
### Description

In the data views, the featured image should provide information about what the image is. Instead, it is actually placed within a button that is labeled with the page title (with the exception of the List layout)

It's worth reminding that an aria label attribute _overrides any content_ so that the button accessible name is actually the page title. The image within the button is completely ignored even when it has an alt text. Basically this markup:

```
<button aria-label="Test page" ...
    <img alt="Farmer boy" ...
    ...
```

does not provide the correct information and I would say not even the correct functionality.

## Table layout

- There's no info at all about the actual image. As said, aria-label overrides any content. As such, the 'Featured image' column actually contains buttons with the post title and nothing else. This defeats the purpose of using a table, where each column should contain the data type identified by the column header.
- I'm not sure why this has to be a button in the first place. Clicking the button switches the editor to edit mode for that page. As a user, this is not what I would expect when clicking the featured image preview of a page.
- Worth noting also the pages without featured image will have a button in the 'Featured image' column. The button is empty and will look like a grey box. It is still labeled with the page title.

Screenshot:

![Screenshot 2024-03-13 at 14 43 55](https://github.com/WordPress/gutenberg/assets/1682452/9bbe410c-baf5-4fde-9d21-26e645c40ef5)

## Grid layout

Similar to the above case. There is no table but the information about the featured image is missing because the button aria-label overrides any content.

## List layout

Similar to the above but the button is actually a div element with a role button `<div role="button"  ...`. It doesn't use an aria-label so that its accessible name is determined by its content: featured image alt text (if any), page title, author and publish status. That's really too much information to label a button, and it gets announced by screen readers all together:

![Screenshot 2024-03-13 at 14 58 25](https://github.com/WordPress/gutenberg/assets/1682452/55ff341b-b947-4438-88e3-cff4e8a31590)

Visually, it would be the equivalent of a button with all that info used as text, which isn't the best way to use a button. Example:

![Screenshot 2024-03-13 at 15 09 15](https://github.com/WordPress/gutenberg/assets/1682452/5ad7916d-acbd-4cd1-a948-e0aa0291ee4d)

### Step-by-step reproduction instructions

- Go to the Site editor > Pages > Manage all pages
- In the View options menu, enable the Featured image field.
- Inspect the source of all the thee Layouts: Table, Grid, and List.
- Observe the featured image is a button and its labeling is less than ideal.

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

**Comment by Rishit30G** [comment]
Thanks for sharing the issue, 

Based on my findings, I noticed that the div element has a role of button for the Grid but not for the List and Table.
Additionally, while aria-labelledby is present, aria-label is missing.
Another issue I observed is that the alt text for the featured image does not appear, even when explicitly provided.

Gutenberg: 20.2.0
WordPress: 6.7.2

https://github.com/user-attachments/assets/dfe1f9a1-ff39-4941-8d40-740dbd1430d4

<img width="1470" alt="Image" src="https://github.com/user-attachments/assets/6c4225d5-66c5-41c5-901f-f8f93f208897" />
